lib/feedback_api_web/facades/survey_create_facade.ex

```diff
@@ -0,0 +1,65 @@
+defmodule FeedbackApiWeb.SurveyCreateFacade do
+  alias FeedbackApi.{Survey, User, Question, Group, Repo}
+  import Ecto.Query
+
+  def create_survey(arguments) do
+    Repo.transaction(fn ->
+      try do
+        survey = Survey.changeset(%Survey{}, arguments) |> Repo.insert!()
+        create_groups(survey, arguments["groups"])
+        create_questions(survey, arguments["questions"])
+      rescue
+        e -> Repo.rollback("Missing required fields")
+      end
+    end)
+  end
+
+  defp create_groups(survey, groups) do
+    Enum.map(groups, fn group -> create_group(survey, group) end)
+  end
+
+  defp create_questions(survey, questions) do
+    Enum.map(questions, fn question -> create_question(survey, question) end)
+  end
+
+  defp create_group(survey, group) do
+    new_group =
+      Group.changeset(%Group{}, group) |> Repo.insert!() |> Repo.preload([:survey, :users])
```

Comment:
What is Repo.preload doing here?

Reply:
It is loading the associated records, which is a requirement to have it build the relationships on the lines below. Otherwise the object will come back without the fields for the associations.